include/mata/afa.hh

```diff
@@ -270,9 +278,20 @@ public:
 
 	StateClosedSet pre(StateClosedSet closed_set) const {return pre(closed_set.antichain());};
 
-	StateClosedSet get_initial_nodes(void) const;
-	StateClosedSet get_non_initial_nodes(void) const;
-	StateClosedSet get_final_nodes(void) const {return StateClosedSet(downward_closed_set, 0, transitionrelation.size()-1, finalstates);};
+	StateClosedSet get_initial_nodes(void) const 
+	{
+		StateClosedSet result = StateClosedSet(upward_closed_set, 0, transitionrelation.size()-1);
+		for(auto node : initialstates)
```

Comment:
`Node` is not a primitive data type. When looping over any set of nodes, you copy the node for each iteration with `auto node: node_set`. Then make another copy inside the loop iteration if you use `insert`, for example. Therefore, I would suggest changing all for loops in the PR to iterate over const references: `const auto& node: node_set`. This way, there will be copies of nodes only when truly necessary. Alternatively, if you need to modify the node (it should not be necessary, if I did not miss anything), you could just remove the `const` keyword to loop over mutable references.

Reply:
Thanks for your suggestion, I appreciate it. I have changed these lines in the code.